akka-stream-tests/src/test/scala/akka/stream/scaladsl/StreamRefsSpec.scala

```diff
@@ -408,4 +415,28 @@ class StreamRefsSpec(config: Config) extends AkkaSpec(config) with ImplicitSende
 
   }
 
+  "The StreamRefResolver" must {
+
+    "serialize and deserialize SourceRefs" in {
+      val probe = TestProbe()
+      val ref = system.actorOf(StreamRefsSpec.SnitchActor.props(probe.ref))
+      val sourceRef = SourceRefImpl[String](ref)
```

Comment:
should we use internal types in the test?

Reply:
That is just to be able to peek into them and verify that the actor was resolved to the same one. Very inside-ious test this.